engine/runtime-integration-tests/src/test/java/org/enso/interpreter/runtime/progress/ProgressTest.java

```diff
@@ -0,0 +1,88 @@
+package org.enso.interpreter.runtime.progress;
+
+import static org.junit.Assert.assertEquals;
+
+import org.enso.common.MethodNames;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.logger.test.TestLogger;
+import org.enso.test.utils.ContextUtils;
+import org.graalvm.polyglot.Context;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+
+public class ProgressTest {
+  private static Context ctx;
+  private static EnsoContext ensoCtx;
+
+  @BeforeClass
+  public static void initCtx() throws Exception {
+    ctx = ContextUtils.createDefaultContext();
+    ensoCtx = ContextUtils.leakContext(ctx);
+  }
+
+  @AfterClass
+  public static void closeCtx() throws Exception {
+    ctx.close();
+    ctx = null;
+  }
+
+  public ProgressTest() {}
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void advanceMultipleTimes() throws Exception {
+    var code =
+        """
+    from Standard.Base import Integer, Float
+    from Standard.Base.Logging import Progress
+
+    geom n:Integer a1:Float q:Float =
+        Progress.run "geometric sequence" n progress->
+            loop i:Integer v:Float acc:Float =
+                if i == n then acc else
+                    progress.log "Step #"+i.to_text
+                    next = v*q
+                    sum = next+acc
+                    progress.advance 1
+                    @Tail_Call loop i+1 next sum
+
+            progress.log "About to compute geometric sequence for "+n.to_text
+            res = loop 1 a1 a1
+            progress.log "We have the result "+res.to_text
+            res
+    """;
+    var log = LoggerFactory.getLogger("Standard.Base.Logging.Progress");
+
+    var geom = ctx.eval("enso", code).invokeMember(MethodNames.Module.EVAL_EXPRESSION, "geom");
+
+    var oneTimeLog =
+        TestLogger.apply(
```

Comment:
To use this `TestLogger` class I created new `logging-test-utils` module in e80dfa00f8eae75c47f74131c47df641b5f26400. That way we can have a regular JPMS dependency on the new module and we don't need to patch/trick the module system to exposes classes originally placed in `test` subdirectory.